### PR TITLE
state: remove deprecated state package

### PR DESCRIPTION
**Pull request by mildwonkey**
REMOVE STATE PACKAGE

There's a lot going on here. It may be easier to review commit-by-commit. 

Here's a summary of the biggest changes:

* The `state` package had a long list of aliases to items in the `statemgr` package. Those have been removed.
* The `state.LocalState` type and functions were specific to the "clistate" (backend config stored in .terraform/terraform.tfstate), so I've moved it into `clistate` (note: my first pass moved it into `terraform`)
* several commits where I removed more aliases I had missed 😝 
* move `state.Remote` into `states.Remote`
* REMOVE STATE PACKAGE 💥

### CODE REVIEW

**Comment by mildwonkey** [comment]
I am going to call this complete: the remaining calls to the `state` package will take (a bit) more thought and effort. The big examples are:

* command/meta_backend.go: the backend state uses the old-style `state` and the functions will need to be refactored
* command/testdata/statelocker: I don't know how much effort this will take, but it wasn't "just" a matter of switching out state for states, so I left it at that for now.

File: backend/local/hook_state.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 // StateHook is a hook that continuously updates the state by calling
-// WriteState on a state.State.
+// WriteState on a statemgr.Full.
```
Comment (pselle) [inline backend/local/hook_state.go:12]: Yay for correcting comments!

**Comment by mildwonkey** [comment]
re-requesting review, since I did A WHOLE BUNCH MORE

**Comment by pselle** [comment]
@mildwonkey Looks like there might still be some dependencies in tests? https://app.circleci.com/pipelines/github/hashicorp/terraform/1380/workflows/cf2f7109-286e-4447-becd-452ae00e0b19/jobs/8818

**Comment by mildwonkey** [comment]
thank you @pselle I hadn't seen that - should be better now 😂

**Pull request merged**